managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -2786,6 +2787,9 @@ public void operationComplete(Void result, Stat stat) {
             @Override
             public void operationFailed(MetaStoreException e) {
                 log.warn("[{}] Failed to update consumer {}", ledger.getName(), name, e);
+                // it means it failed to switch the newly created ledger so, it should be
+                // deleted to prevent leak
+                deleteLedger(lh);
```

Comment:
Here is persistent data to metadata store failed, why need to delete the ledger? We can only delete the ledger after persistent to metadata store successfully?